src/libs/ReportUtils.js

```diff
@@ -4174,41 +4174,52 @@ function getIOUReportActionDisplayMessage(reportAction) {
  * Return room channel log display message
  *
  * @param {Object} reportAction
+ * @param {Boolean} isHtml
  * @returns {String}
  */
-function getChannelLogMemberMessage(reportAction) {
+function getChannelLogMemberMessage(reportAction, isHtml) {
     const verb =
         reportAction.actionName === CONST.REPORT.ACTIONS.TYPE.ROOMCHANGELOG.INVITE_TO_ROOM || reportAction.actionName === CONST.REPORT.ACTIONS.TYPE.POLICYCHANGELOG.INVITE_TO_ROOM
-            ? 'invited'
-            : 'removed';
+            ? Localize.translateLocal('workspace.invite.invited')
+            : Localize.translateLocal('workspace.invite.removed');
 
     const mentions = _.map(reportAction.originalMessage.targetAccountIDs, (accountID) => {
+        if (isHtml) {
+            return `<mention-user accountID=${accountID}></mention-user>`;
+        }
         const personalDetail = lodashGet(allPersonalDetails, accountID);
         const displayNameOrLogin =
             LocalePhoneNumber.formatPhoneNumber(lodashGet(personalDetail, 'login', '')) || lodashGet(personalDetail, 'displayName', '') || Localize.translateLocal('common.hidden');
         return `@${displayNameOrLogin}`;
     });
 
     const lastMention = mentions.pop();
-    let message = '';
+    let message = isHtml ? '<muted-text>' : '';
 
     if (mentions.length === 0) {
         message = `${verb} ${lastMention}`;
     } else if (mentions.length === 1) {
-        message = `${verb} ${mentions[0]} and ${lastMention}`;
+        message = `${verb} ${mentions[0]} ${Localize.translateLocal('common.and')} ${lastMention}`;
     } else {
-        message = `${verb} ${mentions.join(', ')}, and ${lastMention}`;
+        message = `${verb} ${mentions.join(', ')}, ${Localize.translateLocal('common.and')} ${lastMention}`;
     }
 
     const roomName = lodashGet(reportAction, 'originalMessage.roomName', '');
     if (roomName) {
         const preposition =
             reportAction.actionName === CONST.REPORT.ACTIONS.TYPE.ROOMCHANGELOG.INVITE_TO_ROOM || reportAction.actionName === CONST.REPORT.ACTIONS.TYPE.POLICYCHANGELOG.INVITE_TO_ROOM
-                ? ' to'
-                : ' from';
-        message += `${preposition} ${roomName}`;
+                ? ` ${Localize.translateLocal('workspace.invite.to')}`
+                : ` ${Localize.translateLocal('workspace.invite.from')}`;
+
+        const messageHtml = lodashGet(reportAction, ['message', 0, 'html']);
+        const match = messageHtml && messageHtml.match(/<a[^>]*>(.*?)<\/a>/);
```

Comment:
This logic essentially says "let's take the first link from the message's HTML and assume it's a link to the room", right?

Why did you decide to go this route? I thought it would be more consistent to "take over" the responsibility of generating this link ourselves.